test/test_deploy.py

```diff
@@ -0,0 +1,80 @@
+"""Test devnet contract deployment"""
```

Comment:
I see that these are unit tests, but would it make sense to move here the salty_deploy and failing_deploy tests from test_general_workflow.py?